cloudify_types/cloudify_types/component/polling.py

```diff
@@ -184,20 +183,19 @@ def is_deployment_execution_at_state(client,
     execution = client.executions.get(execution_id=execution_id,
                                       _include=execution_get_args)
     ctx.logger.info(
-        'Execution "{0}" of component "{1}" state is {2}'.format(execution_id,
-                                                                 dep_id,
-                                                                 execution))
+        'Execution "%s" of component "%s" state is %s',
+        execution_id, dep_id, execution)
 
     if log_redirect:
         ctx.logger.debug(
-            'Execution info with log_redirect is {0}'.format(execution))
+            'Execution info with log_redirect is %s', execution)
```

Comment:
Why format will not work here?

Reply:
format does work but in general this form is more correct and I wish we were using it everywhere... I'm not sure whether it's better for new changes to keep consistent with existing code, or use the better form.

The difference is that, note that this isn't actually doing the formatting right here (the `execution` is a separate argument to `ctx.logger.debug`). This way, the logging mechanism only formats the value into the string if the log will actually be emitted. If the logging level is set so that debug messages aren't shown, string formatting (which is potentially an expensive operation) won't be done this way